Xamarin.Forms.Core/PlatformConfiguration/iOSSpecific/Page.cs

```diff
@@ -60,10 +60,14 @@ public static UIStatusBarAnimation PreferredStatusBarUpdateAnimation(this IPlatf
 
 		public static readonly BindableProperty UseSafeAreaProperty = BindableProperty.Create("UseSafeArea", typeof(bool), typeof(Page), false, propertyChanged: (bindable, oldValue, newValue) =>
 		{
+			if (Device.RuntimePlatform != Device.iOS) return;
+
 			var page = bindable as Xamarin.Forms.Page;
 			if ((bool)oldValue && !(bool)newValue)
 			{
-				page.Padding = default(Thickness);
+				var safeAreaInsets = GetSafeAreaInsets(page);
+				if (safeAreaInsets == page.Padding)
+					page.Padding = default(Thickness);
```

Comment:
not related to your change, but directly setting the `Padding` from the SafeAreaInsets looks a bad idea. we probably need something like an `ActualPadding` property that sum the padding with the insets

Reply:
Agree, i think what we decided at the time was like if you are setting padding you just don't use SafeAreInsets, if you are using padding you will be responsabile to make sure on iOS to you get the insets and add that t your padding. that's why we expose the Insets by itself also.

Reply:
Agree, I figured from the implementation that it was setup to be all or nothing and not additive. My other thought is that we are going to have to redo this for the Android notch at which point I feel like this code will just move directly into the APIs of the Forms controls opposed to being a platform thing at which point we could make all this a little bit better.

For example detecting rotation and things like that